packages/core/src/lib/implementation/execute-plugin.ts

```diff
@@ -123,15 +125,50 @@ export async function executePlugins(
 
     progressBar?.updateTitle(`Executing  ${chalk.bold(pluginCfg.title)}`);
 
-    const pluginReport = await executePlugin(pluginCfg);
-    progressBar?.incrementInSteps(plugins.length);
-
-    return outputs.concat(pluginReport);
-  }, Promise.resolve([] as PluginReport[]));
+    try {
+      const pluginReport = await executePlugin(pluginCfg);
+      progressBar?.incrementInSteps(plugins.length);
+      return outputs.concat(Promise.resolve(pluginReport));
+    } catch (e: unknown) {
+      progressBar?.incrementInSteps(plugins.length);
+      return outputs.concat(
+        Promise.reject(e instanceof Error ? e.message : String(e)),
+      );
+    }
+  }, Promise.resolve([] as Promise<ExecutePluginResult>[]));
 
   progressBar?.endProgress('Done running plugins');
 
-  return pluginsResult;
+  const errorsCallback = ({ reason }: PromiseRejectedResult) =>
+    console.error(reason);
+  const results = await Promise.allSettled(pluginsResult);
+
+  logMultipleResults(results, 'Plugins', undefined, errorsCallback);
+
+  // TODO: add groupBy method for promiseSettledResult by status, #287
+
+  const failedResults = results.filter(
+    (
+      result: PromiseSettledResult<ExecutePluginResult>,
+    ): result is PromiseRejectedResult => result.status === 'rejected',
+  );
+  if (failedResults.length) {
+    const errorMessages = failedResults
+      .map(({ reason }: PromiseRejectedResult) => reason)
+      .join(', ');
+    throw new Error(
+      `Plugins failed: ${failedResults.length} errors: ${errorMessages}`,
+    );
+  }
+
+  return results
+    .filter(
+      (
+        result: PromiseSettledResult<ExecutePluginResult>,
+      ): result is PromiseFulfilledResult<PluginReport> =>
+        result.status === 'fulfilled',
+    )
+    .map((result: PromiseFulfilledResult<PluginReport>) => result.value);
```

Comment:
The type guards I suggest could be useful here as well, I see :grin: 

```suggestion
  const failedResults = results.filter(isPromiseRejectedResult);
  if (failedResults.length) {
    const errorMessages = failedResults
      .map(({ reason }) => reason)
      .join(', ');
    throw new Error(
      `Plugins failed: ${failedResults.length} errors: ${errorMessages}`,
    );
  }

  return results
    .filter(isPromiseFulfilledResult)
    .map(result => result.value);
```

Reply:
Done. It really helps both for DRY and readability